### PR TITLE
Remove code filtering out operation none children from BoundBlock

### LINKED ISSUE #21776: IBlockStatement should not filter out all OperationKind.None
Currently, when we create `IBlockStatement`s, we filter out all `OperationKind.None` operations. We should not do this, and figure out why this is there.

### PR DESCRIPTION
**Pull request by mavasani**
Fixes #21776

### CODE REVIEW

**Review by 333fred — APPROVED** [review]

**Comment by mavasani** [comment]
@dotnet-bot retest windows_release_vs-integration_prtest please

**Comment by mavasani** [comment]
@dotnet/roslyn-compiler for one more review

@jinujoseph for approval

**Comment by jinujoseph** [comment]
Approved to merge for 15.8.Preview5

File: src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory.vb [hunk 1]
```diff
@@ -1370,6 +1364,16 @@ Namespace Microsoft.CodeAnalysis.Operations
...
         End Function
 
+        Private Function CreateBoundLocalDeclarationOperation(boundLocalDeclaration As BoundLocalDeclaration) As IVariableDeclarationGroupOperation
+            Dim declarations As Lazy(Of ImmutableArray(Of IVariableDeclarationOperation)) =
+                New Lazy(Of ImmutableArray(Of IVariableDeclarationOperation))(Function() GetVariableDeclarationStatementVariables(ImmutableArray.Create(Of BoundLocalDeclarationBase)(boundLocalDeclaration)))
+            Dim syntax As SyntaxNode = boundLocalDeclaration.Syntax
+            Dim type As ITypeSymbol = Nothing
+            Dim constantValue As [Optional](Of Object) = New [Optional](Of Object)()
+            Dim isImplicit As Boolean = boundLocalDeclaration.WasCompilerGenerated
+            Return New LazyVariableDeclarationGroupOperation(declarations, _semanticModel, syntax, type, constantValue, isImplicit)
```
Comment (AlekseyTs) [inline src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory.vb]: >isImplicit [](start = 120, length = 10)

Should this node be always implicit since we are reusing the same syntax for it?

Comment (mavasani) [reply src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory.vb]: I made the binder change in the PR to ensure `BoundLocalDeclaration` generated for this code path is always marked as compiler generated. Are you suggesting asserting that here or in the test visitors?

Comment (AlekseyTs) [reply src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory.vb]: I am suggesting to always pass true, because regardless how we mark BoundLocalDeclaration, this is going to be the second node for the same syntax.

---
In reply to: [198340134](https://github.com/dotnet/roslyn/pull/28135#discussion_r198340134) [](ancestors = 198340134)

**Comment by mavasani** [comment]
@dotnet-bot retest ubuntu_16_mono_debug_prtest please

**Comment by mavasani** [comment]
@dotnet/roslyn-compiler for an additional review, given Aleksey is OOF.

**Comment by mavasani** [comment]
@dotnet-bot retest windows_debug_vs-integration_prtest please

File: src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory.vb [hunk 2]
```diff
@@ -1281,15 +1283,7 @@ Namespace Microsoft.CodeAnalysis.Operations
         Private Function CreateBoundBlockOperation(boundBlock As BoundBlock) As IBlockOperation
             Dim statements As Lazy(Of ImmutableArray(Of IOperation)) = New Lazy(Of ImmutableArray(Of IOperation))(
                 Function()
-                    ' We should not be filtering OperationKind.None statements.
```
Comment (jcouv) [inline src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory.vb]: >OperationKind.None [](start = 49, length = 18)

I expected to see some tests with additional None operations in the printout, but I didn't see any. Is it that we don't have any way of producing None operation nowadays?

**Review by jcouv — APPROVED** [review]
LGTM (iteration 2) with one question about test coverage. Thanks

File: src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory.vb (hunk 2 above)
Comment (mavasani) [reply src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory.vb]: @jcouv We do still produce none operations for NYI bound nodes. However, this change is just removing a piece of code that was dropping node operations that were direct children of bound block.

Comment (mavasani) [reply src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory.vb]: We also have tests for these none operations. For example, you can search for `OperationKind.None` in the baselines for http://source.roslyn.io/#q=IOperationTests_IParameterReferenceExpression
